json-schema-core/src/main/java/io/ballerina/jsonschema/core/SchemaExceptions.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com)
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.jsonschema.core;
+
+/**
+ * Custom Schema Exceptions.
+ *
+ * @since 0.1.0
+ */
+public class SchemaExceptions {
+    public static class InvalidJsonSchemaException extends Exception {
+        public InvalidJsonSchemaException(String message) {
+            super(message);
+        }
+
+        public InvalidJsonSchemaException(String message, Throwable cause) {
+            super(message, cause);
+        }
+    }
+
+    public static class EmptyJsonSchemaException extends Exception {
+        public EmptyJsonSchemaException(String message) {
+            super(message);
+        }
+
+        public EmptyJsonSchemaException(String message, Throwable cause) {
+            super(message, cause);
+        }
+    }
```

Comment:
We don't need separate exceptions for each exception scenario, just a JsonSchemaToolException should do.